doc/api/crypto.md

```diff
@@ -2111,6 +2132,71 @@ Based on the recommendations of [NIST SP 800-131A][]:
 
 See the reference for other recommendations and details.
 
+### CCM mode
+
+CCM is one of the two supported [AEAD algorithms][]. Applications which use this
+mode must adhere to certain restrictions when using the cipher API:
+
+- The authentication tag length must be specified during cipher creation by
+  setting the `authTagLength` option and must be one of 4, 6, 8, 10, 12, 14 or
+  16 bytes.
+- The length of the initialization vector (nonce) `N` must be between 7 and 13
+  bytes (`7 ≤ N ≤ 13`).
+- The length of the plaintext is limited to `2^(8 * (15 - N))` bytes.
+- When decrypting, the authentication tag must be set via `setAuthTag()` before
+  specifying additional authenticated data and / or calling `update()`.
+  Otherwise, decryption will fail and `final()` will throw an error in
+  compliance with section 2.6 of [RFC 3610][].
+- It is not recommended to use stream methods such as `write(data)`, `end(data)`
+  or `pipe()` in CCM mode.
```

Comment:
What happens when you do?

Reply:
The main issue here is that `update()` can only be called once, so the CCM cipher is not very stream-like, and using it as a duplex stream might give the impression that writing multiple chunks of data will work. Maybe @shigeki has further insights, he suggested to add this note in https://github.com/nodejs/node/pull/18138#discussion_r166205704.

Reply:
In that case I’d say something along the lines of `Using stream methods such as … will fail, because only one chunk of data per instance will be handled by these objects.` here? How does that sound?

Like, I’d just say that it doesn’t work instead of saying it’s not recommended. That doesn’t seem very far from the truth. ;)